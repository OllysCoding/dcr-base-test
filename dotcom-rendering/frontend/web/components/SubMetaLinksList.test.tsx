import React from 'react';
import { shallow } from 'enzyme';
import { SubMetaLinksList } from './SubMetaLinksList';

describe('SubMetaLinksList', () => {
    const links: SimpleLinkType[] = [
        {
            url: '/test/1',
            title: 'Test 1',
        },
        {
            url: '/test/2',
            title: 'Test 2',
        },
    ];
    const isSectionLinkList: boolean = false;
    const pillar: Pillar = 'news';

    describe('snapshots', () => {
        it('It should render correctly if isSectionLinkList true', () => {
            const component = shallow(
                <SubMetaLinksList
                    links={links}
                    isSectionLinkList={true}
                    pillar={pillar}
                />,
            );

            expect(component).toMatchSnapshot();
        });

        it('It should render correctly if isSectionLinkList false', () => {
            const component = shallow(
                <SubMetaLinksList
                    links={links}
                    isSectionLinkList={false}
                    pillar={pillar}
                />,
            );

            expect(component).toMatchSnapshot();
        });
    });

    it('It should render correct amount of links', () => {
        const component = shallow(
            <SubMetaLinksList
                links={links}
                isSectionLinkList={isSectionLinkList}
                pillar={pillar}
            />,
        );

        expect(component.find('li').length).toBe(2);
    });
});
