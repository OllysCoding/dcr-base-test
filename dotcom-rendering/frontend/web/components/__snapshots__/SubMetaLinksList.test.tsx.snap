// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubMetaLinksList snapshots It should render correctly if isSectionLinkList false 1`] = `
.emotion-0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #c70000;
}

.emotion-0:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-4 {
  list-style: none;
  padding-bottom: 12px;
  margin-bottom: 6px;
  border-bottom: 1px solid #dcdcdc;
}

.emotion-1 {
  margin-right: 5px;
  display: inline-block;
  font-size: 14px;
  line-height: 22px;
  font-family: GuardianTextSans,Helvetica Neue,Helvetica,Arial,Lucida Grande,sans-serif;
}

.emotion-1 a {
  position: relative;
  display: block;
  padding-right: 5px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-1 a::after {
  content: '/';
  position: absolute;
  pointer-events: none;
  top: 0;
  right: -3px;
  color: #dcdcdc;
}

.emotion-1 a:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-3 {
  margin-right: 5px;
  display: inline-block;
  font-size: 14px;
  line-height: 22px;
  font-family: GuardianTextSans,Helvetica Neue,Helvetica,Arial,Lucida Grande,sans-serif;
}

.emotion-3 a {
  position: relative;
  display: block;
  padding-right: 5px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-3 a::after {
  content: '/';
  position: absolute;
  pointer-events: none;
  top: 0;
  right: -3px;
  color: #dcdcdc;
}

.emotion-3 a:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-3 a::after {
  content: '';
}

<ul
  className="emotion-4"
>
  <li
    className="emotion-1"
    key="/test/1"
  >
    <a
      className="emotion-0"
      href="/test/1"
    >
      Test 1
    </a>
  </li>
  <li
    className="emotion-3"
    key="/test/2"
  >
    <a
      className="emotion-0"
      href="/test/2"
    >
      Test 2
    </a>
  </li>
</ul>
`;

exports[`SubMetaLinksList snapshots It should render correctly if isSectionLinkList true 1`] = `
.emotion-4 {
  list-style: none;
  font-weight: 500;
  line-height: 24px;
}

.emotion-1 {
  margin-right: 5px;
  display: inline-block;
  font-size: 16px;
  line-height: 20px;
  font-family: GuardianTextEgyptian,Georgia,serif;
}

.emotion-1 a {
  position: relative;
  display: block;
  padding-right: 5px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-1 a::after {
  content: '/';
  position: absolute;
  pointer-events: none;
  top: 0;
  right: -3px;
  color: #dcdcdc;
}

.emotion-1 a:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #c70000;
}

.emotion-0:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-3 {
  margin-right: 5px;
  display: inline-block;
  font-size: 16px;
  line-height: 20px;
  font-family: GuardianTextEgyptian,Georgia,serif;
}

.emotion-3 a {
  position: relative;
  display: block;
  padding-right: 5px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-3 a::after {
  content: '/';
  position: absolute;
  pointer-events: none;
  top: 0;
  right: -3px;
  color: #dcdcdc;
}

.emotion-3 a:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-3 a::after {
  content: '';
}

<ul
  className="emotion-4"
>
  <li
    className="emotion-1"
    key="/test/1"
  >
    <a
      className="emotion-0"
      href="/test/1"
    >
      Test 1
    </a>
  </li>
  <li
    className="emotion-3"
    key="/test/2"
  >
    <a
      className="emotion-0"
      href="/test/2"
    >
      Test 2
    </a>
  </li>
</ul>
`;
